packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -508,7 +559,15 @@ export function createRegularIndex(
       await createFn(ns, spec, options);
       dispatch(indexCreationSucceeded(inProgressIndexId));
       track('Index Created', trackEvent, connectionInfoRef.current);
-
+      // See action description for details
+      if (isRollingIndexBuild) {
+        setTimeout(() => {
+          dispatch({
+            type: ActionTypes.RollingIndexTimeoutCheck,
+            indexId: inProgressIndexId,
+          });
+        }, POLLING_INTERVAL * 3);
+      }
```

Comment:
how long does it normally take for the rolling index to be created? 15s doesn't sound like much

Reply:
It's a bit hard to wrap the head around (tried to do it [here](https://github.com/mongodb-js/compass/pull/6331/files#diff-150603fed00664cb5992ebff90acae175937dded2d5c1ca1d16628c5eb8471b8R105-R113) but not sure if I did a good job to be honest), but we don't need to wait for the index to be created before this timeout check kicks in, we are only waiting for the rolling index build to start showing up in the list returned from the backend which corresponds to the index starting to be created on any of the nodes in the cluster, which happens way faster. As soon as it starts showing up in the list returned from the rolling indexes service, it will be removed from the in progress list and this action firing will be a no-op basically

Reply:
ah I see, thanks!